types/yup/index.d.ts

```diff
@@ -212,18 +212,29 @@ export type TransformFunction<T> = ((
     originalValue: any
 ) => any);
 
-export interface WhenOptionsBuilder<T> {
+export interface WhenOptionsBuilderFunction<T> {
     (value: any, schema: T): T;
     (v1: any, v2: any, schema: T): T;
     (v1: any, v2: any, v3: any, schema: T): T;
     (v1: any, v2: any, v3: any, v4: any, schema: T): T;
 }
 
-export type WhenOptions<T> =
-    | WhenOptionsBuilder<T>
-    | { is: boolean | ((...values: any[]) => boolean); then: any; otherwise: any }
+export type WhenOptionsBuilderObjectIs =
```

Comment:
Since `is` can actually be anything, are we losing of the type information if we just change this type to `((...values: any[]) => boolean) | any` rather than a union of all these types?

Furthermore, can we name this type something like `BuilderIsProperty` or `WhenIsProperty` or simply `IsProperty` so that it doesn't sound like a ruby method name?

Reply:
Yes. Unioning anything with `any` resolves to just `any`. I believe this is even a linter warning. The way I have it allows for all the types I tested with, but still gives some intellisense/direction about what the `when` method expects.

I think `IsProperty` is too generic. What if there were another `is` property that took different types. Likewise with `BuilderIsProperty`. There could be other methods that take a builder of a different sort. Again, I was trying to stick with and expand the existing naming convention.

Reply:
I'm still not a fan of the name but I don't see that as a reason to block this PR from being merged.